modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -533,10 +545,39 @@ def downsample_trigger_training_set(self) -> None:
         self._train_dataloader.dataset.change_key_source(new_key_source)
 
         self.update_queue(AvailableQueues.DOWNSAMPLING, batch_number, number_of_samples, training_active=True)
+        # set the model to train
+        self._model.model.train()
+
+    def _iterate_dataloader_and_compute_scores(
+        self,
+        dataloader: torch.utils.data.DataLoader,
+        previous_batch_number: int = 0,
+        previous_number_of_samples: int = 0,
```

Comment:
could you add some documentation why this is needed? it appears some per class vs non per class thing but its not obvious to me what is happening here

Reply:
Sure